engine/Shopware/Plugins/Default/Core/CronProductExport/Bootstrap.php

```diff
@@ -79,17 +79,18 @@ public function exportProductFiles()
             /** @var Shopware\Models\ProductFeed\ProductFeed $feedModel */
             $fileName = $feedModel->getHash() . '_' . $feedModel->getFileName();
             $filePath = $cacheDir . $fileName;
+
             if ($feedModel->getInterval() === 0) {
                 continue;
-            }elseif($feedModel->getInterval() > 0) {
-                $diffInterval = time();
-                if ($feedModel->getCacheRefreshed()) {
-                    $diffInterval = $diffInterval - $feedModel->getCacheRefreshed()->getTimestamp();
-                }
+            }
 
-                if ($diffInterval < $feedModel->getInterval() && file_exists($filePath)) {
-                    continue;
-                }
+            $diffInterval = time();
+            if ($feedModel->getCacheRefreshed()) {
+                $diffInterval -= $feedModel->getCacheRefreshed()->getTimestamp();
+            }
+
+            if ($diffInterval < $feedModel->getInterval() && file_exists($filePath)) {
+                continue;
```

Comment:
hallo @tinect ,

thanks for your changes, but I think I will stick with the original Code.

the elseif condition hier is required to handle the 'only cron' option ( interval = -1 ), and by this option the code must always create the feed file by every request.

about calculate the diffInterval I just use the the same code hier:

https://github.com/shopware/shopware/blob/5.6/engine/Shopware/Controllers/Backend/Export.php#L83

Regard's
Ahmad

Reply:
Hey @ahmad-saad,
ahhh, didn't see that fact. Thank you.